testsuite/ada_lsp/V628-024.symbol_diff/test.json

```diff
@@ -0,0 +1,275 @@
+                  "text": "with Ada.Text_IO;\n\nprocedure Main is\n   procedure Print (I : Integer; B : Boolean) is\n   begin\n      Ada.Text_IO.Put_Line (\"Hello\" & B'Img & I'Img);\n   end;\n\nI : constant Integer := 0;\nB : constant Boolean := False;\nbegin\n   Print (I, B);\nend Main;\n"
+               }
+            }
+         },
+         "wait": []
+      }
+   },
+   {
+      "send": {
+         "request": {
+            "jsonrpc": "2.0",
+            "method": "textDocument/didChange",
+            "params": {
+               "textDocument": {
+                  "uri": "$URI{main.adb}",
+                  "version": 1
+               },
+               "contentChanges": [
+                  {
+                     "range": {
+                        "start": {
+                           "line": 8,
+                           "character": 26
+                        },
+                        "end": {
+                           "line": 8,
+                           "character": 26
+                        }
+                     },
+                     "text": "\n"
+                  }
+               ]
+            }
+         },
+         "wait": []
+      }
+   },
+   {
+      "send": {
+         "request": {
+            "jsonrpc": "2.0",
+            "id": 5,
+            "method": "textDocument/rangeFormatting",
+            "params": {
+               "textDocument": {
+                  "uri": "$URI{main.adb}"
+               },
+               "range": {
+                  "start": {
+                     "line": 8,
+                     "character": 0
+                  },
+                  "end": {
+                     "line": 10,
+                     "character": 1
+                  }
+               },
+               "options": {
+                  "tabSize": 3,
+                  "insertSpaces": true,
+                  "trimTrailingWhitespace": true,
+                  "insertFinalNewline": false,
+                  "trimFinalNewlines": true
+               }
+            }
+         },
+         "wait": [
+            {
+               "id": 5,
+               "result": [
+                  {
+                     "range": {
+                        "start": {
+                           "line": 8,
+                           "character": 0
+                        },
+                        "end": {
+                           "line": 8,
+                           "character": 0
+                        }
+                     },
+                     "newText": "   "
+                  },
+                  {
+                     "range": {
+                        "start": {
+                           "line": 9,
+                           "character": 0
+                        },
+                        "end": {
+                           "line": 9,
+                           "character": 0
+                        }
+                     },
+                     "newText": "\u0000"
```

Comment:
I wouldn't join this patch before fixing insertion of binary zero into the text. In my opinion, it could break editors including GNAT Studio.

Reply:
I've already tested with GNAT Studio and there are not any issues with this.

Reply:
Where this comes from?

Reply:
From gnatpp

Reply:
The test fails also, needs to checked. @CKMonika is the "\u0000" that comes from gnatpp expected?

Reply:
It's strange, I don't see any reason for that... Maybe @reznikmm has an idea?

Reply:
Actually I think the test is failing due to a timing issue: the test seems to be stopped while indexing the project. You are probably missing a wait somewhere

Reply:
> It's strange, I don't see any reason for that... Maybe @reznikmm has an idea?

because Andry tries to modify `master` while `edge` has its-own modification for range formatting and doesn't use diff any more:

```ada
   Partial_GNATpp : constant GNATCOLL.Traces.Trace_Handle :=
     GNATCOLL.Traces.Create ("ALS.PARTIAL_GNATPP",
                             GNATCOLL.Traces.On);
   --  Use partial formatting mode of gnatpp if On. Otherwise, use diff
   --  algorithm.
```